File: 알고리즘/김연우/JAVA/2670.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        double[] arr = new double[n+1];
        double[] dp = new double[n+1];
        double max = 0;
        for (int i = 1; i <= n; i ++){
            arr[i] = Double.parseDouble(br.readLine());
            dp[i] = Double.max(arr[i], dp[i-1] * arr[i]);
            max = Math.max(max, dp[i]);
        }
        System.out.println(String.format("%.3f", max));
    }

}

File: 알고리즘/유찬홍/백준/02000/2670.c
#include <stdio.h>

int main() {
    int n;
    double arr[10001], max = 0;
    scanf("%d", &n);
    for (int i = 0; i < n; i++) {
        scanf("%lf", &arr[i]);
    }
    for (int i = 0; i < n; i++) {
        double rs = 1;
        for (int j = i; j < n; j++) {
            rs *= arr[j];
            if (max < rs) max = rs;
        }
    }
    printf("%.3lf", max);
}
